File: 251001/Smychok/LAB1/Models/Issue.cs
namespace LAB1.Models;

public class Issue
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Content { get; set; }
    public DateTime Created { get; set; }
    public DateTime Modified { get; set; }
    public int WriterId { get; set; }
    public Writer Writer { get; set; }
    public ICollection<Sticker> Stickers { get; set; } = new List<Sticker>();
    public ICollection<Post> Posts { get; set; } = new List<Post>();
}

File: 251003/zdanovich/Publisher/src/main/java/by/zdanovich/Publisher/models/Issue.java
package by.zdanovich.Publisher.models;


import jakarta.persistence.*;
import lombok.Data;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "tbl_issue")
@Data
public class Issue {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne()
    @JoinColumn(name = "writerId", referencedColumnName = "id")
    private Writer writer;

    @Column(name = "title", unique = true)
    private String title;

    @Column(name = "content")
    private String content;

    @Column(name = "created")
    private Date created;

    @Column(name = "modified")
    private Date modified;

    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(name = "tbl_issue_mark", joinColumns = @JoinColumn(name = "issue_id"),
            inverseJoinColumns = @JoinColumn(name = "mark_id"))
    private List<Mark> marks = new ArrayList<>();

}
